webapp/channels/src/components/post_markdown/post_markdown.tsx

```diff
@@ -53,106 +57,203 @@ export type OwnProps = {
+    const postChannelTeamId = useSelector((state: GlobalState) => {
+        if (!post) {
+            return null;
         }
+        const postChannel = getChannel(state, post.channel_id);
+        return postChannel?.team_id || null;
+    });
 
-        if (this.props.post && this.props.post.type === Posts.POST_TYPES.WRANGLER) {
-            const renderedWranglerMessage = renderWranglerSystemMessage(this.props.post);
-            return <div>{renderedWranglerMessage}</div>;
+    // Extract channel mentions from post and determine which ones are missing
+    const {mentionsToFetch} = useMemo(() => {
+        if (!post) {
+            return {mentionsToFetch: []};
         }
 
-        if (this.props.post && this.props.post.type === PostTypes.CUSTOM_DATA_SPILLAGE_REPORT) {
-            return (
-                <div>
-                    <DataSpillageReport
-                        post={this.props.post}
-                        isRHS={this.props.isRHS}
-                    />
-                </div>
+        // Extract all mentions from post content (message + attachments)
+        const allMentions = extractChannelMentionsFromPost(post);
+
+        // Filter to find mentions not in redux store
+        const missing = allMentions.filter((channelName) => {
+            // Check if channel exists in redux store by name
+            const channelExists = Object.values(allChannels).some(
+                (ch) => ch.name.toLowerCase() === channelName && ch.team_id === postChannelTeamId,
             );
+
+            return !channelExists;
+        });
+
+        return {mentionsToFetch: missing};
+    }, [post, allChannels, postChannelTeamId]);
+
+    // Get teams for building channel names map
+    const allTeams = useSelector((state: GlobalState) => state.entities.teams.teams);
+    const postTeam = useSelector((state: GlobalState) => {
+        if (!postChannelTeamId) {
+            return null;
         }
+        return state.entities.teams.teams[postChannelTeamId] || null;
+    });
 
-        // Proxy images if we have an image proxy and the server hasn't already rewritten the this.props.post's image URLs.
-        const proxyImages = !this.props.post || !this.props.post.message_source || this.props.post.message === this.props.post.message_source;
-        const channelNamesMap = isChannelNamesMap(this.props.post?.props?.channel_mentions) ? this.props.post?.props?.channel_mentions : undefined;
+    // Fetch missing channels
+    useEffect(() => {
+        if (mentionsToFetch.length === 0 || !postTeam) {
+            return;
+        }
 
-        this.props.pluginHooks?.forEach((o) => {
-            if (o && o.hook && this.props.post) {
-                message = o.hook(this.props.post, message);
+        // Dispatch fetch for each missing channel
+        mentionsToFetch.forEach((channelName) => {
+            dispatch(getChannelByNameAndTeamName(postTeam.name, channelName, true) as any).catch((error: Error) => {
```

Comment:
Note that this will result in potentially many requests being made by the app at once, especially if you switch to a channel with many of these posts in it. Ideally, we would add an API to fetch multiple channels by name at once to avoid that, but that can be added in the future.